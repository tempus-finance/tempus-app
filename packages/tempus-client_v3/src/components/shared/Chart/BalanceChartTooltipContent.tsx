import { FC } from 'react';
import { Ticker } from 'tempus-core-services';
import FormattedDate from '../FormattedDate';
import Typography from '../Typography';
import './Chart.scss';

export interface BalanceChartTooltipContentProps {
  title?: string;
  value: string;
  currency: Ticker | 'USD';
  date: Date;
  locale: string;
}

const BalanceChartTooltipContent: FC<BalanceChartTooltipContentProps> = props => {
  const { title, value, currency, date, locale } = props;

  return (
    <div className="tc__chart__balance-tooltip-content">
      {title && (
        <Typography
          className="tc__chart__balance-tooltip-content-title"
          variant="body-primary"
          weight="bold"
          color="text-secondary"
        >
          {title}
        </Typography>
      )}
      <div className="tc__chart__balance-tooltip-content-value">
        <Typography variant="subtitle" type="mono" weight="bold">
          {value} {currency}
        </Typography>
      </div>
      <FormattedDate date={date} locale={locale} size="small" textColor="text-secondary" separatorContrast="low" />
    </div>
  );
};

export default BalanceChartTooltipContent;
