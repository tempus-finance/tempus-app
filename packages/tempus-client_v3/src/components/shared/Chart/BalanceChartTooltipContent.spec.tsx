import { render } from '@testing-library/react';
import BalanceChartTooltipContent, { BalanceChartTooltipContentProps } from './BalanceChartTooltipContent';

const defaultProps: BalanceChartTooltipContentProps = {
  title: 'Deposit',
  value: '0.123',
  currency: 'ETH',
  date: new Date(Date.UTC(2022, 4, 1)),
};

const subject = (props: BalanceChartTooltipContentProps) => render(<BalanceChartTooltipContent {...props} />);

describe('BalanceChartTooltipContent', () => {
  it('renders a balance tooltip content', () => {
    const { container } = subject(defaultProps);

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
