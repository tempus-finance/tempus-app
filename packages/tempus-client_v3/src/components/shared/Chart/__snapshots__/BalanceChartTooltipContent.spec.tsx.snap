// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceChartTooltipContent renders a balance tooltip content 1`] = `
<div>
  <div
    class="tc__chart__balance-tooltip-content"
  >
    <div
      class="tc__chart__balance-tooltip-content-title"
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(71, 67, 80); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Deposit
    </div>
    <div
      class="tc__chart__balance-tooltip-content-value"
    >
      <div
        style="font-style: normal; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'Azeret Mono', monospace; opacity: 1;"
      >
        0.123
         
        ETH
      </div>
    </div>
    <span
      class="tc__formatted-date"
    >
      <div
        style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
      >
        May
      </div>
      <span
        class="tc__formatted-date__separator tc__formatted-date__separator-low-contrast"
      />
      <div
        style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
      >
        01
      </div>
      <span
        class="tc__formatted-date__separator tc__formatted-date__separator-low-contrast"
      />
      <div
        style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
      >
        22
      </div>
    </span>
  </div>
</div>
`;
